src/Domain/CreateInitialMesh.hpp

```diff
@@ -14,9 +14,8 @@
 #include "NumericalAlgorithms/Spectral/Spectral.hpp"
 #include "Utilities/Gsl.hpp"
 
-/// Items for initializing the DataBox%es of parallel components
-namespace Initialization {
-/// \ingroup InitializationGroup
+namespace domain {
```

Comment:
I don't think this should be moved out of the Initialization namespace and InitializationGroup.  This function is only to be used in the Initialization phase.  This is also true for `create_initial_element` , so I would argue it should be moved into the Initialization namespace and InitializationGroup.